web/docs/deployment/ci-cd.md

```diff
@@ -1,7 +1,113 @@
+1. Install Wasp in the CI environment.
+2. Run your app (with the database) in the CI environment.
+3. Run the e2e tests against the running app.
+
+#### Example app
+
+We'll show you how to run end-to-end tests in CI using the [Github Actions](https://github.com/features/actions) as our CI and the [Playwright](https://playwright.dev/) as our e2e testing framework.
+
+1. Check our example app and its e2e tests in the [e2e-tests](https://github.com/wasp-lang/e2e-test-example/tree/main/e2e-tests) directory. 
+
+    You can copy the `e2e-tests` directory to your own project and modify it to fit your app. This will enable you to run the e2e tests locally.
+    
+    <details>
+    <summary>Example e2e test</summary>
+
+    ```ts
+    import { expect, test } from "@playwright/test";
+    import { generateRandomUser, logUserIn } from "./utils";
+
+    const user = generateRandomUser();
+
+    test.describe("basic user flow test", () => {      
+      test("log in and add task", async ({ page }) => {
+        await logUserIn({ page, user });
+        await expect(page).toHaveURL("/");
+        await expect(page.locator("body")).toContainText("No tasks yet.");
+
+        // Add a task
+        await page.fill('input[name="description"]', "First task");
+        await page.click('input:has-text("Create task")');
+        await expect(page.locator("body")).toContainText("First task");
+      });
+    });
+    ```
+    </details>
+2. To run the tests in the Github Actions CI, you'll need to create a workflow file in your repository.
+
+    You should create a `.github/workflows/e2e-tests.yml` file in your repository. You can copy the contents of the [e2e-tests.yml](https://github.com/wasp-lang/e2e-test-example/blob/main/.github/workflows/e2e-tests.yml) file from our example app.
+    
+### Unit tests
+
+Unit tests test pieces of your code logic in isolation. They are much simpler and faster than e2e tests, but they don't simulate the real user interaction with your app.
+
+You can use Wasp's built in [client tests](../project/testing.md) support to test the client side code of your app. You are free to use any testing framework for the server side code.
+
+**You'd run the unit tests in the CI** in a similar way as the e2e tests:
+1. Install Wasp in the CI environment.
+2. Run the client tests with `wasp test client run`.
+3. Run the server tests with your testing framework.
+
+## Continuous Deployment using Docker
+
+The most common way to package your app for deployment is using Docker images. This way you can easily deploy the same image to different environments (staging, production, etc.).
+
+**To build the app as a Docker image**, you'll need to:
+1. Install Docker in the CI environment.
+2. Build the app with `wasp build`.
+3. Build the Docker image and push it to a Docker registry:
+    - for our server app
+    - for our client app
+4. Notify your deployment provider to deploy the new image.
```

Comment:
This can depend on their hostirng provider, right? If I am correct some of them have their own registries so this looks more like just a single command, or maybe you just give them a branch with Dockerfile, stuff like that?
Also client app doesn't have to be in Dockerfile.
Also might be worth  mentiong here that we provide Dockerfile for the server.

And now that I am writnig all this, sounds to me like this is quite complex and we should instead point them to the part of deployment docs that describes it in details? Maybe just be very simple here, try not to assume much, link to the part with dpeloying Wasp app, and say that they will want to use whatever method their hosting providers reocmmends for CD. And an example deployment of ours, that is cool.